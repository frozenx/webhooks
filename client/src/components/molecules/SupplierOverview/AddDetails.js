import React from "react";
import { IconButtonLinkWithText } from '@beans/link';
import { AddDetailsWrapper } from "./styled";

const AddDetails = props => {
    return (
        <AddDetailsWrapper>
            <IconButtonLinkWithText
                icon={{ graphic: 'add', size: 'xs', position: { global: 'left' } }}
                buttonVariant="secondary"
                disabled={!props.canAddNewSite}
                emphasized={true}
                onClick={() => { props.addNewSite() }}
            >
                {props.title}
            </IconButtonLinkWithText>
        </AddDetailsWrapper>
    );
}

export default AddDetails;
