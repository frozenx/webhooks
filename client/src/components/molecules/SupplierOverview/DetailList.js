import React from "react";
import { StandaloneLinkWithIcon, StandaloneLink } from '@beans/link';
import { ListWrapper } from "./styled";

const DetailList = props => {
    const {
        attributeGroups,
        handleSupplierForm,
        handleUUID,
        uuid,
        attributeGroupIds,
        node
    } = props;
    return (
        <ListWrapper>
            <ul>
                {node.fields && node.fields.map((attrGrp, index) => {
                    let {
                        id: formId,
                        name
                    } = attrGrp;
                    const isViewMode = attrGrp.action === 'enabled';
                    return (
                        <li className='formGroup' onClick={handleUUID(uuid)} key={index}>
                            <StandaloneLink disabled={!isViewMode} emphasized={true} variant="standalone">{name}</StandaloneLink>
                            <StandaloneLinkWithIcon
                                className='add-icon'
                                icon={{ graphic: 'forwardLink', position: { global: 'right' } }}
                                emphasized={true}
                                onClick={() => handleSupplierForm(attrGrp, node)}
                                disabled={!isViewMode && node.attributeRules[node.type][attrGrp.key] && !node.attributeRules[node.type][attrGrp.key].create}
                            > {isViewMode ? 'View' : 'Add'}
                            </StandaloneLinkWithIcon>
                        </li>
                    )
                })}
            </ul>
        </ListWrapper>
    );
}

export default DetailList;
