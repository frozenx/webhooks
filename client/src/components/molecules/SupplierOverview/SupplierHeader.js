import React, { Fragment } from 'react';
import { FormHeading, TopHeading } from '../../../common/styles';
import Label from '@beans/label';
import { HeaderOverview, StyledList, SupplierOverviewContainer } from './styled';

const SupplierHeader = props => {
  const {
    partnerName,
    staticAttributes,
    values
  } = props;
  return (
    <SupplierOverviewContainer>
      <FormHeading>
        <TopHeading><h1>{partnerName}</h1></TopHeading>
      </FormHeading>
      <HeaderOverview>
        <Label dark={true} emphasized={true} className="supplier-title">
          {' '}
          Partner details
        </Label>
        <StyledList>
          {
            staticAttributes.map((staticAttribute, i) => {
              return (
                <li key={`partner-static-attribute-${i}`}>
                  <Label dark={true} emphasized={true}>
                    {staticAttribute.displayName}
                  </Label>
                  <Label>{values[staticAttribute.key]}</Label>
                </li>
              )
            })
          }
        </StyledList>
      </HeaderOverview>
    </SupplierOverviewContainer>
  );
};

export default SupplierHeader;
