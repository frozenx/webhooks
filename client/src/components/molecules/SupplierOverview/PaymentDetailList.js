import React from "react";
import { StandaloneLinkWithIcon, StandaloneLink } from '@beans/link';
import AddDetails from './AddDetails';
import { ListWrapper } from "./styled";

const DetailList = props => {
    const {
        attributeGroups,
        handleSupplierForm,
        handleUUID,
        uuid,
        attributeGroupIds,
        node,
        addPayment,
        index
    } = props;
    const isViewMode = node.fields[0].action === 'enabled'; 
    return (
    <React.Fragment>
        <li className='formGroup' onClick={handleUUID(uuid)} key={index}>
            <StandaloneLink disabled={!isViewMode} emphasized={true} variant="standalone">{`Payment terms ${index + 1}`}</StandaloneLink>
            <StandaloneLinkWithIcon
                className='add-icon'
                icon={{ graphic: 'forwardLink', position: { global: 'right' } }}
                emphasized={true}
                onClick={() => handleSupplierForm(node.fields[0], node)}
                    disabled={!isViewMode && node.attributeRules[node.type][node.fields[0]] && !node.attributeRules[node.type][node.fields[0].key].create}
            > {isViewMode ? 'View' : 'Add'}
            </StandaloneLinkWithIcon>
        </li>

    </React.Fragment>
               
                    


               

      
    );
}

export default DetailList;
