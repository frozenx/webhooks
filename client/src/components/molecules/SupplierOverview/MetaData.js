import * as utils from '../../pages/SupplierHeader/utils';


class Meta {
    static presets = {};
    constructor({ uuid, fields, type, version, values, errors, parentUuid, attributeRules = {}, partnerStaticAttributes = [], addressBook, contactBook }) {
        this.attributeRules = attributeRules;
        this.uuid = uuid;
        this.fields = fields;
        this.type = type;
        this.values = values;
        this.errors = errors;
        this.version = version;
        this.children = [];
        this.partnerStaticAttributes = partnerStaticAttributes;
        this.isChildEnabled = this.isChildEnabled.bind(this);
        this.isSiblingFilled = this.isSiblingFilled.bind(this);
        this.initializeValues = this.initializeValues.bind(this);
        this.addChild = this.addChild.bind(this);
        this.updateFields = this.updateFields.bind(this);
        this.parentUuid = parentUuid;
        this.addressBook = addressBook;
        this.contactBook = contactBook;
        this.initializeValues(this.fields);

    }
    isSiblingFilled() {
        const childrenLength = this.children.length;
        return childrenLength
            && this.children.every(child => child.fields[0].action === 'enabled');
    }
    isChildEnabled(siblingCheck = true) {
        const fieldCheck = this.fields.every(field => field.action === 'enabled');
        if (!this.children.length || !siblingCheck) {
            return fieldCheck;
        }
        return fieldCheck && this.isSiblingFilled();
    }
    addChild(node, siblingCheck = true) {
        if (!this.isChildEnabled(siblingCheck)) {
            console.log('Sorry cannot add need to fill fields')
            return false;
        } else {
            console.log('child added with id - ', node.id);
            node.parentUuid = this.uuid;
            const newNode = new Meta(node);
            this.children.push(newNode)
            return newNode;
        }

    }
    updateFields(fields) {
        this.fields = fields;
    }
    update({ uuid, version }) {
        this.uuid = uuid;
        this.version = version;
    }
    updateNode(info) {
        this.uuid = info.uuid;
        this.version = info.version;
        Object.keys(info.values).forEach((attributeKey) => {

            this[attributeKey] = { values: { ...info.values[attributeKey] || {} }, errors: {} };
        });
        this.fields = this.fields.map((field) => {
            const attributeKeys = Object.keys(this[field.key].values)
            if ((this[field.key] && attributeKeys.length && this.isAllRequiredAttrAreFilled(field, attributeKeys)) ) {
                attributeKeys.forEach((attributeKey) => {
                    this.attributeRules = {...this.attributeRules}
                    this.attributeRules[this.type] = utils.updateAttributeRules(Meta.presets, this.attributeRules[this.type], attributeKey, this[field.key].values[attributeKey], this.type || '');
                });
                return { ...field, action: 'enabled' }
            }

            return { ...field };
        });
    }
    initializeValues() {
        this.fields.forEach((field) => {
            this[field.key] = { values: {}, errors: {} };
        });
    }
    getValues(currentlyEnteredValues, key) {
        const areNodeAndFormKeysSame = (nodeKey) => key === nodeKey
        return this.fields.reduce((acc, field) => {
            return this[field.key] ? {
                ...acc,
                [field.key]: areNodeAndFormKeysSame(field.key) ?
                    { ...currentlyEnteredValues[field.key].values } : { ...this[field.key].values }
            } : acc
        }, {})
    }
    clearValues(key) {
        const updatedFields = this.fields.map(field => {
            if (field.key === key) {
                return { ...field, action: 'disabled' }
            }
            return { ...field };
        });
        this.updateFields(updatedFields);
        this[key] = { values: {}, errors: {} };
    }

    isAllRequiredAttrAreFilled(attrGrp, attributeKeys){
        const {key, collectionType} = attrGrp;
        if(!collectionType){
            const requiredAttributes = Object.keys(this.attributeRules[this.type]).filter(attr => this.attributeRules[this.type][attr].required && this.attributeRules[this.type][attr].attrGrpKey === key)
            const filledAttrKeys = [...attributeKeys];
            return requiredAttributes.every(attr => filledAttrKeys.indexOf(attr) > -1)
        }
         return true;
    }
}


export default Meta;