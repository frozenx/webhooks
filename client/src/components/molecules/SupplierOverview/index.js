import React, { Component } from "react";
import Notification from '@beans/notification';
import {
  FormHeading,
  SecondaryWrapper,
  StyleGrayWrapper,
  NotificationStyle,
  PrimaryWrapper
} from "../../../common/styles";
import DetailList from './DetailList';
import PaymentDetailList from './PaymentDetailList';
import SupplierHeader from "./SupplierHeader";
import Label from '@beans/label';
import AppConstants from "../../../languagepack";
import { SupplierStepTwo, HeaderListWrapper, SiteListWrapper, ListWrapper } from "./styled";
import AddDetails from './AddDetails';
import scroller from '../../../lib/scroller';

class SupplierOverview extends Component {

  constructor(props) {
    super(props);
    this.state = {};
  }

  componentWillMount() {
    scroller();
  }

  getSuccessTitle = (partnerName, currentFieldName) => {
    return `${partnerName} ${currentFieldName} have been successfully updated`;
  }


  _handleSupplierForm = (selectedAttributeGroup, node) => {
    const { handleStepThree } = this.props
    handleStepThree(selectedAttributeGroup, node)
  }

  _handleUUID = (uuid) => {
    //  console.log(uuid)
  }

  render() {
    const {
      metaData: {
        attributeGroups = [],
        partnerAttributeStatic = {}
      },
      siteMetaData = {},
      paymentMetaData = {},
      attributeGroupIds,
      supplierHeaderData,
      supplierHeaderData: {
        uuid,
        partnerAttributes: {
          partnerName = ''
        }
      },
      rootNode,
      isFormSubmissionSuccessful,
      currentFieldName
    } = this.props;

    return (
      <SupplierStepTwo>
        <SupplierHeader
          staticAttributes={rootNode.partnerStaticAttributes}
          partnerName={(rootNode.partner.values || {}).partnerName || ''}
          values={rootNode.partner.values} />
        <PrimaryWrapper>
          <SecondaryWrapper>
            {isFormSubmissionSuccessful &&
              <NotificationStyle>
                <Notification title={this.getSuccessTitle(rootNode.partner.values.partnerName, currentFieldName)} variant={'success'} />
              </NotificationStyle>
            }
            <FormHeading>
              <h1>{`${rootNode.partner.values.partnerName ? rootNode.partner.values.partnerName : ''} ${AppConstants.supplierDetails}`}</h1>
              <h4>{AppConstants.stepTwoSubTitle}</h4>
            </FormHeading>

            <HeaderListWrapper>
              <DetailList
                attributeGroups={attributeGroups}
                uuid={rootNode.uuid}
                handleUUID={this._handleUUID}
                handleSupplierForm={this._handleSupplierForm}
                attributeGroupIds={attributeGroupIds}
                node={rootNode} />
            </HeaderListWrapper>

            <SiteListWrapper>
              {
                rootNode.children.map((node, nodeIndex) =>
                  <StyleGrayWrapper key={nodeIndex}>
                    <Label
                      dark={true}
                      emphasized={true}
                      className="site-title"> {`${rootNode.partner.values.partnerName} ${AppConstants.siteDetails}`} </Label>
                    <DetailList
                      attributeGroups={siteMetaData.attributeGroups}
                      uuid={rootNode.uuid}
                      handleUUID={this._handleUUID}
                      handleSupplierForm={this._handleSupplierForm}
                      attributeGroupIds={attributeGroupIds}
                      node={node} />

                    <StyleGrayWrapper>
                      <Label
                        dark={true}
                        emphasized={true}
                        className="payment-title"> {AppConstants.paymentDetails}
                      </Label>
                      {
                        node.children.map((paymentNode, index) =>
                          <>
                            <ListWrapper showBorderTop={index}>
                              <ul>
                                <PaymentDetailList
                                  attributeGroups={paymentMetaData.attributeGroups}
                                  uuid={rootNode.uuid}
                                  handleUUID={this._handleUUID}
                                  handleSupplierForm={this._handleSupplierForm}
                                  attributeGroupIds={attributeGroupIds}

                                  node={paymentNode}
                                  index={index}
                                  addPayment={this.props.addPayment} />
                              </ul>
                            </ListWrapper>
                          </>
                        )
                      }
                      <AddDetails
                        title={'Add payment terms'}
                        description={'Payment terms'}
                        addNewSite={() => this.props.addPayment(node)}
                        canAddNewSite={node.isChildEnabled()} />
                    </StyleGrayWrapper>
                  </StyleGrayWrapper>
                )
              }

              {<StyleGrayWrapper>
                <Label
                  dark={true}
                  emphasized={true}
                  className="site-title"> {AppConstants.siteDetails} </Label>
                <AddDetails
                  title={'Add site details'}
                  description={'Site details'}
                  addNewSite={() => this.props.addSite(rootNode)}
                  canAddNewSite={rootNode.isChildEnabled()} />
              </StyleGrayWrapper>}
            </SiteListWrapper>

          </SecondaryWrapper>
        </PrimaryWrapper>
      </SupplierStepTwo>
    );
  }
}

export default SupplierOverview;
