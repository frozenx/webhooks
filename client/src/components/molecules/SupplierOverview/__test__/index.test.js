jest.mock('../../../../lib/httpClient');
import React from 'react';
import { shallow } from 'enzyme';
import SupplierOverview from '../index';
import httpClient from '../../../../lib/httpClient';

describe('supplier overview component test cases', () => {
  let Wrapper;
  let mockProps = 
  {
    supplierHeaderData : {
      uuid: '1234'
    },
    partnerName: 'Uni',
    handleStepThree: jest.fn()
  }

  beforeEach(() => {
    Wrapper = shallow(<SupplierOverview {...mockProps}/>);
  });

  it('render the component', () => {
    expect(Wrapper).toMatchSnapshot();
  });
  test('should call the supplier header Api on componentDidMount', () => {  
    const instance = Wrapper.instance();
    jest.spyOn(instance, '_callSupplierHeaderApi');
    instance.componentDidMount();  
    expect(instance._callSupplierHeaderApi).toHaveBeenCalled();
  });
  test('should give a response greater than 400', () => {
    httpClient.get.mockResolvedValue({
      status: 404,
      message: 'error'
    })
    try{
      const instance = Wrapper.instance();
      jest.spyOn(instance, '_callSupplierHeaderApi');
      instance.componentDidMount();
    } catch(error) {
      expect(error.message).toBe('error')
    }            
  })
  test('should give a response less than 400',  async (done) => {
    httpClient.get.mockResolvedValue({
      status: 200,
    })
    const instance = Wrapper.instance();
    jest.spyOn(instance, '_callSupplierHeaderApi');
    instance.componentDidMount();
    await Wrapper.update();
    setTimeout(() => {
      expect(Wrapper.state().displayLoader).toBe(false);
      done()
    },0)
  })

  test('should call the method', () => {
    Wrapper.setState({supplierData: { attributeGroups: [{id: 1}]}})
    const instance = Wrapper.instance();
    jest.spyOn(instance, '_handleSupplierForm');
    instance._handleSupplierForm(1);
    expect(mockProps.handleStepThree).toHaveBeenCalled();

  })
  
});
