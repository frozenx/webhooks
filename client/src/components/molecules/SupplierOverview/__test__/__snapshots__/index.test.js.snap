// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supplier overview component test cases render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SupplierOverview
    handleStepThree={[MockFunction]}
    partnerName="Uni"
    supplierHeaderData={
      Object {
        "uuid": "1234",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SupplierHeader
          partnerName="Uni"
          supplierHeaderData={
            Object {
              "uuid": "1234",
            }
          }
        />,
        <ForwardRef>
          <ForwardRef>
            <h1>
              Uni supplier details
            </h1>
            <h4>
              Create a new supplier by entering the details below
            </h4>
          </ForwardRef>
          <ForwardRef>
            <DetailList
              attributeGroups={Array []}
              handleSupplierForm={[Function]}
              handleUUID={[Function]}
              uuid="1234"
            />
          </ForwardRef>
          <ForwardRef>
            <ForwardRef>
              <AddDetails />
            </ForwardRef>
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "partnerName": "Uni",
          "supplierHeaderData": Object {
            "uuid": "1234",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <h1>
                Uni supplier details
              </h1>
              <h4>
                Create a new supplier by entering the details below
              </h4>
            </ForwardRef>,
            <ForwardRef>
              <DetailList
                attributeGroups={Array []}
                handleSupplierForm={[Function]}
                handleUUID={[Function]}
                uuid="1234"
              />
            </ForwardRef>,
            <ForwardRef>
              <ForwardRef>
                <AddDetails />
              </ForwardRef>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <h1>
                  Uni supplier details
                </h1>,
                <h4>
                  Create a new supplier by entering the details below
                </h4>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Uni supplier details",
                },
                "ref": null,
                "rendered": "Uni supplier details",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create a new supplier by entering the details below",
                },
                "ref": null,
                "rendered": "Create a new supplier by entering the details below",
                "type": "h4",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "rules": Array [
                  [Function],
                  ";",
                  "-webkit-tap-highlight-color:",
                  [Function],
                  ";border:0;box-sizing:border-box;margin:0;outline:",
                  [Function],
                  ";padding:0;position:relative;",
                  ";",
                  "
    margin-top: 34px;
    .input-error {
        color: #cc3333;
    }
",
                ],
              },
              "displayName": "Styled(base-components__RootElement)",
              "foldedComponentIds": Array [
                "base-components__RootElement-sc-150tvch-1",
              ],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                null,
                <DetailList
                  attributeGroups={Array []}
                  handleSupplierForm={[Function]}
                  handleUUID={[Function]}
                  uuid="1234"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "attributeGroups": Array [],
                  "handleSupplierForm": [Function],
                  "handleUUID": [Function],
                  "uuid": "1234",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-jzJRlG",
                "isStatic": true,
                "rules": Array [
                  "
    margin-bottom: 24px;
    position: relative;
    .loader{
        background-color: transparent;
        position: static;
        > div{
            width: 6em;
            height: 6em;
        }
    }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-jzJRlG",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef>
                <AddDetails />
              </ForwardRef>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <AddDetails />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-kAzzGY",
                  "isStatic": true,
                  "rules": Array [
                    "
    background-color: #f6f6f6;
    border: solid 1px #e5e5e5;
    padding: 32px 105px 40px 105px;
    margin-bottom: 24px;
    label.site-title{
        font-size: 24px;
        margin-bottom: 20px
    }
    @media (max-width: 756px) {
       padding-left: 63px;
       padding-right: 63px;
    }  
    @media (max-width: 505px) {
        padding-left: 42px;
        padding-right: 42px;
    } 
    @media (max-width: 320px) {
        padding-left: 12px;
        padding-right: 12px;
    } 
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-kAzzGY",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-cSHVUG",
                "isStatic": true,
                "rules": Array [
                  "
    
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-cSHVUG",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              [Function],
              ";",
              "-webkit-tap-highlight-color:",
              [Function],
              ";border:0;box-sizing:border-box;margin:0;outline:",
              [Function],
              ";padding:0;position:relative;",
              ";",
              "
    width: 816px;
    margin: 0 auto;
    @media (max-width: 816px) {
        width: 100%;
    }
    form{
        padding-bottom: 24px;
    }
",
            ],
          },
          "displayName": "Styled(base-components__RootElement)",
          "foldedComponentIds": Array [
            "base-components__RootElement-sc-150tvch-1",
          ],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jTzLTM",
        "isStatic": true,
        "rules": Array [
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-jTzLTM",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SupplierHeader
            partnerName="Uni"
            supplierHeaderData={
              Object {
                "uuid": "1234",
              }
            }
          />,
          <ForwardRef>
            <ForwardRef>
              <h1>
                Uni supplier details
              </h1>
              <h4>
                Create a new supplier by entering the details below
              </h4>
            </ForwardRef>
            <ForwardRef>
              <DetailList
                attributeGroups={Array []}
                handleSupplierForm={[Function]}
                handleUUID={[Function]}
                uuid="1234"
              />
            </ForwardRef>
            <ForwardRef>
              <ForwardRef>
                <AddDetails />
              </ForwardRef>
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "partnerName": "Uni",
            "supplierHeaderData": Object {
              "uuid": "1234",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <h1>
                  Uni supplier details
                </h1>
                <h4>
                  Create a new supplier by entering the details below
                </h4>
              </ForwardRef>,
              <ForwardRef>
                <DetailList
                  attributeGroups={Array []}
                  handleSupplierForm={[Function]}
                  handleUUID={[Function]}
                  uuid="1234"
                />
              </ForwardRef>,
              <ForwardRef>
                <ForwardRef>
                  <AddDetails />
                </ForwardRef>
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <h1>
                    Uni supplier details
                  </h1>,
                  <h4>
                    Create a new supplier by entering the details below
                  </h4>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Uni supplier details",
                  },
                  "ref": null,
                  "rendered": "Uni supplier details",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create a new supplier by entering the details below",
                  },
                  "ref": null,
                  "rendered": "Create a new supplier by entering the details below",
                  "type": "h4",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    [Function],
                    ";",
                    "-webkit-tap-highlight-color:",
                    [Function],
                    ";border:0;box-sizing:border-box;margin:0;outline:",
                    [Function],
                    ";padding:0;position:relative;",
                    ";",
                    "
    margin-top: 34px;
    .input-error {
        color: #cc3333;
    }
",
                  ],
                },
                "displayName": "Styled(base-components__RootElement)",
                "foldedComponentIds": Array [
                  "base-components__RootElement-sc-150tvch-1",
                ],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  null,
                  <DetailList
                    attributeGroups={Array []}
                    handleSupplierForm={[Function]}
                    handleUUID={[Function]}
                    uuid="1234"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "attributeGroups": Array [],
                    "handleSupplierForm": [Function],
                    "handleUUID": [Function],
                    "uuid": "1234",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-jzJRlG",
                  "isStatic": true,
                  "rules": Array [
                    "
    margin-bottom: 24px;
    position: relative;
    .loader{
        background-color: transparent;
        position: static;
        > div{
            width: 6em;
            height: 6em;
        }
    }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-jzJRlG",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef>
                  <AddDetails />
                </ForwardRef>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <AddDetails />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-kAzzGY",
                    "isStatic": true,
                    "rules": Array [
                      "
    background-color: #f6f6f6;
    border: solid 1px #e5e5e5;
    padding: 32px 105px 40px 105px;
    margin-bottom: 24px;
    label.site-title{
        font-size: 24px;
        margin-bottom: 20px
    }
    @media (max-width: 756px) {
       padding-left: 63px;
       padding-right: 63px;
    }  
    @media (max-width: 505px) {
        padding-left: 42px;
        padding-right: 42px;
    } 
    @media (max-width: 320px) {
        padding-left: 12px;
        padding-right: 12px;
    } 
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-kAzzGY",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-cSHVUG",
                  "isStatic": true,
                  "rules": Array [
                    "
    
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-cSHVUG",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                [Function],
                ";",
                "-webkit-tap-highlight-color:",
                [Function],
                ";border:0;box-sizing:border-box;margin:0;outline:",
                [Function],
                ";padding:0;position:relative;",
                ";",
                "
    width: 816px;
    margin: 0 auto;
    @media (max-width: 816px) {
        width: 100%;
    }
    form{
        padding-bottom: 24px;
    }
",
              ],
            },
            "displayName": "Styled(base-components__RootElement)",
            "foldedComponentIds": Array [
              "base-components__RootElement-sc-150tvch-1",
            ],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-jTzLTM",
          "isStatic": true,
          "rules": Array [
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-jTzLTM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
